---
title: "Integrating Schizophrenia and Height Genome Wide CNV Studies"
output:
  html_document:
    df_print: paged
---


A short notebook to compare and contrast the CNV profiles and locations for the schizophrenia and 
height genome wide CNV studies.

Both the schizophrenia and height CNV databases that have been downloaded are in hg18, so I don't 
need to convert for these operations, so if I want to decide which genes the CNVs are overlapping then
I need to use hg18.


## Preparation


```{r prepare, echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
library(here)
source(here("R", "prepare.R"))
install.load("pander", "ggplot2", "data.table", "gridExtra", "reshape2")#, "gatepoints")
install.load.bioc("GenomicRanges") 

## Read the non-gene centric results.
scz.dup <- fread(dropbox("pgc_cnv/PGC_41K_QC_dup.cnv.results"))
scz.del <- fread(dropbox("pgc_cnv/PGC_41K_QC_del.cnv.results"))

gscz.dup <- GRanges(seqnames=scz.dup$CHR, IRanges(start=scz.dup$BP, width=1), 
                    z=scz.dup$z, p=scz.dup$z_pval, f=scz.dup$NCNV/41321, 
                    fcontrol =scz.dup$UNAFF/20227 , n=scz.dup$NCNV, fcase = scz.dup$AFF/21094)

gscz.del <- GRanges(seqnames=scz.del$CHR, IRanges(start=scz.del$BP, width=1), z=scz.del$z, 
                    p=scz.del$z_pval, f=scz.del$NCNV/41321, 
                    fcontrol =scz.del$UNAFF/20227 , n=scz.del$NCNV, fcase = scz.del$AFF/21094)

genome(gscz.del) <- "hg18"
genome(gscz.dup) <- "hg18"

height_cnv <- fread(dropbox("height_CNV_association_41467_2017_556_MOESM2_ESM.csv"))

gheight_cnv <- GRanges(height_cnv$CHR, IRanges(height_cnv$BP, width=1), 
                       p_height = height_cnv$`Pvalue Height`, beta_height= height_cnv$`Beta Height`, 
                       p_bmi = height_cnv$`Pvalue BMI`, beta_bmi = height_cnv$`Beta BMI`, 
                       fdel=height_cnv$F_DEL, fdup = height_cnv$F_DUP)
genome(gheight_cnv) <- "hg18"
rm(scz.dup, scz.del, height_cnv)  ##keep the workspace tidy

```

We have `r length(gscz.del)` rows of deletions and `r length(gscz.dup)` rows of duplications for 
schizophrenia and `r length(gheight_cnv)` cnv locations in total for the for the height study.

#### Schizophrenia

The schizophrenia data are based on cohort of 21,094 cases and 20,227 controls, giving a total sample 
size of 41,321. These were :

> *A final set of rare, highquality CNVs was defined as those >20 kb in length, encompassing at
> least 10 probes and <1% minor allele frequency (MAF). *

#### Height

> *Depending on the trait, the sample sizes varied between 161,244 and 191,161.*

There seems to be a bit of a  problem with the height CNVs.  A proportion of these 
have frequencies greater than 1.   This is the same for the database frequencies that are on the 
web site.  I'll just ignore it for now (I imagine this is because some cnvs have multiple 
copies so this is a product of the frequency and the number of copies).  More likely than this is 
that the height CNVs are given as a percentage not as a raw frequency.

Plot the frequencies in cases and controls.  Bold points are those significant at $p<0.001$.

```{r scz_frequencies, cache=TRUE, echo=FALSE}
scz <- rbind( 
  data.frame(case=gscz.del$fcase, control=gscz.del$fcontrol, p=gscz.del$p, type="del"),
  data.frame(case=gscz.dup$fcase, control=gscz.dup$fcontrol, p=gscz.dup$p, type="dup")
)
pl <- ggplot(scz, aes(x=control, y=case, col=type)) + geom_point(alpha=0.02) + guides(colour = guide_legend(override.aes = list(alpha = 1))) 
scz2 <- scz[scz$p<0.001,]
pl <- pl + geom_point(data=scz2, aes(x=control, y=case, col=type), shape=21, fill="white")
pl
rm(scz2)

```

### Tabulate frequencies

```{r prepare_table_freq, echo=FALSE}
scz$p <- NULL
s <- reshape2::melt(scz, id.vars="type", variable.name="cohort", value.name="freq" )
s$cat <- cut(s$freq, breaks=c(0,0.0001,0.0002,0.0005,0.0010,0.0020,0.0050,0.0100,1000) )
tb <- table(cat = s$cat, cohort=s$cohort, type=s$type)
#pander(tb)
tbf <- as.data.frame(tb)
#pander(tbf)
```

### Plot Frequencies

```{r plot_freq, echo=FALSE}
ggplot(tbf, aes(y=Freq, x=cat, col=cohort)) + geom_point(alpha=0.5) +facet_wrap(~type)

rm(tb, tbf, scz, s)
```

Overall there appears to be little difference in the frequencies in the cases and controls for duplications and deletions.

## Overlaps between Schizophrenia and height

### Duplications

#### Preparation

Look at the near CNVs that are at closest in the schizohrenia and height datasets.


```{r create_comparison_datasets, echo=FALSE, warning=FALSE}
nearest.scz.dup <- nearest(gheight_cnv, gscz.dup)    ## find the nearest duplication
height_scz.dup <- cbind(data.frame(gheight_cnv), data.frame(gscz.dup[nearest.scz.dup]) )
## Tidy the combined dataset
height_scz.dup$strand <- NULL;height_scz.dup$strand <- NULL
colnames(height_scz.dup)[c(2,12,16)] <- c("start_h","start_s","p_s")

height_scz.dup$close <- abs(height_scz.dup$start_h - height_scz.dup$start_s)<100

height_scz.dup$distance <- cut(abs(height_scz.dup$start_h - height_scz.dup$start_s), c(0, 100, 10000, 100000,100000000), include.lowest=TRUE)

ggplot(height_scz.dup, aes(x=fdup/100, y=fcontrol)) + geom_point(alpha=0.02) + xlim(0,0.012) + ylim(0,0.012) + xlab("Height Duplication CNV Frequency") + ylab("SCZ Duplication Frequency (controls)") +ggtitle("Duplication Frequencies") + guides(colour = guide_legend(override.aes = list(alpha = 1))) + facet_wrap(~distance) + geom_smooth(method="lm"  )

```


It is clear that by matching close CNVs we get better agreement in the CNV frequencies.  How many CNVs do we get in the four distance classes?

```{r table_scz_matches_dup, echo=FALSE}
pander(table(distance = height_scz.dup$distance), caption="How close are the duplication matches?")
```
I can cut down the number of positions to look at, and get a set of CNVs that more cosely match the frequencies in 
both datasets by just choosing those within 100 bases.  This leaves `r sum (height_scz.dup$close)` CNVs.




### Deletions 


```{r scz_deletions_preparation, echo=FALSE, warning=FALSE}

nearest.scz.del <- nearest(gheight_cnv, gscz.del)
height_scz.del <- cbind(data.frame(gheight_cnv), data.frame(gscz.del[nearest.scz.del]) )
height_scz.del$strand <- NULL;height_scz.del$strand <- NULL

colnames(height_scz.del)[c(2,12,16)] <- c("start_h","start_s","p_s")

height_scz.del$close <- abs(height_scz.del$start_h - height_scz.del$start_s)<100


height_scz.del$distance <- cut(abs(height_scz.del$start_h - height_scz.del$start_s), c(0, 100, 10000, 100000,100000000), include.lowest=TRUE)

ggplot(height_scz.del, aes(x=fdel/100, y=fcontrol)) + geom_point(alpha=0.02) + xlim(0,0.012) + ylim(0,0.012) + xlab("Height Deletion CNV Frequency") + ylab("SCZ Deletion in controls") +ggtitle("Deletion") + guides(colour = guide_legend(override.aes = list(alpha = 1))) + facet_wrap(~distance) + geom_smooth(method="lm")

```


```{r deletion_distance_table}
pander(table(distance=height_scz.del$distance))
```


Let's also try to join the deletions and duplications together for the scz data

```{r tryjoin, echo=FALSE}

nearest.scz <- nearest(gscz.del, gscz.dup)
all.scz <- cbind(data.frame(gscz.del), data.frame(gscz.dup[nearest.scz]) )[,-c(3:5,12,14:16)]

colnames(all.scz)[9:15] <- paste(colnames(all.scz)[2:8],"dup", sep="_")
colnames(all.scz)[2:8] <- paste(colnames(all.scz)[2:8],"del", sep="_")


all.scz$col <- factor((all.scz$p_dup<0.01)  +2*(all.scz$p_del<0.01), labels=c("NS", "dup sig", "del sig", "Both sig"))
all.scz$distance <- cut(abs(all.scz$start_del - all.scz$start_dup), c(0,1, 1000, 10000,1000000000), include.lowest=TRUE)

pander(table(all.scz$distance), caption="Distance from deletion CNV to0 nearest duplication CNV")
ggplot(all.scz[all.scz$distance=="[0,1]",], aes(y=z_del, x=z_dup, col=col)) + geom_point(alpha=0.1)
all.scz <- all.scz[all.scz$distance=="[0,1]",]
attach(all.scz)
gall.scz <- GRanges(seqnames=all.scz$seqnames, IRanges(all.scz$start_del, width=1), 
                    zdel=z_del, zdup=z_dup,
                    pdel=p_del, pdup=p_dup,
                    fcase_del=fcase_del, fcase_dup=fcase_dup,
                    fcontrol_del=fcontrol_del, fcontrol_dup=fcontrol_dup,
                    n_del=n_del, n_dup=n_dup)
genome(gall.scz) <- "hg18"
detach(all.scz)
rm(all.scz)
```

For a majority of the locations I could just use the deletions and duplications together.


## Analysis

### Duplication

```{r dup_analysis, echo=FALSE}
height_scz.dup <- height_scz.dup[height_scz.dup$close, ]   #use just those that are close
pander(table(cut(height_scz.dup$start_h-height_scz.dup$start_s, breaks=c(-100,-50,-10,-1,0,1,10,50,100))), caption="Difference in Position between Height and SCZ CNV")
```
#ggplot(height_scz.dup, aes(x=start_h-start_s)) + geom_histogram()
```{r dup_plots, echo=FALSE}
height_scz.dup$h_col <- factor((height_scz.dup$p_s<0.01)  +2*(height_scz.dup$p_h<0.01), labels=c("NS", "S sig", "H sig", "H&S sig"))
height_scz.dup$bmi_col <- factor((height_scz.dup$p_s<0.01)  +2*(height_scz.dup$p_bmi<0.01), labels=c("NS", "S sig", "BMI sig", "BMI&S sig"))
height_scz.dup$cut_cnvfreq <- factor(cut(height_scz.dup$fdel-height_scz.dup$fdup, c(-100,-0.01,0.01,100)), labels = c("del+", "equal", "dup+"))
height_scz.dup$raredel <- factor(height_scz.dup$fdel/100<0.0002, levels=c(T,F), labels=c("rare del", "common del"))
height_scz.dup$raredup <- factor(height_scz.dup$fdup/100<0.0002, levels=c(TRUE, FALSE), labels=c("rare dup", "common dup"))



ggplot(height_scz.dup, aes(x=beta_height, y=z, col=h_col)) + geom_point(alpha=0.1) + theme(legend.title=element_blank()) + guides(colour = guide_legend(override.aes = list(alpha = 1))) + ggtitle("Effect Sizes for Height vs SCZ") + geom_point(data=height_scz.dup[height_scz.dup$hcol!="NS",], aes(x=beta_height, y=z, col=h_col), alpha=1)

if (FALSE) {
  p1 <- ggplot(height_scz.dup, aes(x=fcontrol, y=fcase, col=h_col)) + geom_point(alpha=0.05) + theme(legend.title=element_blank()) + guides(colour = guide_legend(override.aes = list(alpha = 1))) 
  p2 <-ggplot(height_scz.dup, aes(x=fcontrol, y=fcase, col=bmi_col)) + geom_point(alpha=0.05) + theme(legend.title=element_blank()) + guides(colour = guide_legend(override.aes = list(alpha = 1))) 
  grid.arrange(p1, p2, ncol=2, top=)
}


ggplot(height_scz.dup, aes(x=fcase, y=fcontrol, col=bmi_col)) + geom_point(alpha=0.05) + theme(legend.title=element_blank()) + guides(colour = guide_legend(override.aes = list(alpha = 1))) +facet_wrap(~cut_cnvfreq) #, scales="free")



#table(height_scz.dup$cut_cnvfreq)
table(height_scz.dup$raredel, height_scz.dup$raredup)


ggplot(height_scz.dup, aes(x=beta_height, y=z, col=h_col)) + geom_point(alpha=0.05) + theme(legend.title=element_blank()) + guides(colour = guide_legend(override.aes = list(alpha = 1))) + facet_grid(raredel~raredup) 

ggplot(height_scz.dup, aes(x=beta_bmi, y=z, col=bmi_col)) + geom_point(alpha=0.05) + theme(legend.title=element_blank()) + guides(colour = guide_legend(override.aes = list(alpha = 1))) + facet_grid(raredel~raredup) 

```


### Deletion

```{r analyse_deletion}
height_scz.del <- height_scz.del[height_scz.del$close,]
ggplot(height_scz.del, aes(x=start_h-start_s)) + geom_histogram()


height_scz.del$h_col <- factor( (height_scz.del$p_s<0.01)  +2*(height_scz.del$p_h<0.01), labels=c("NS", "S sig", "H sig", "H&S sig"))
height_scz.del$bmi_col <- factor( (height_scz.del$p_s<0.01)  +2*(height_scz.del$p_bmi<0.01), labels=c("NS", "S sig", "BMI sig", "BMI&S sig"))
height_scz.del$cut_cnvfreq <- factor(cut(height_scz.del$fdel-height_scz.del$fdup, c(-100,-0.01,0.01,100)), labels = c("del+", "equal", "dup+"))


ggplot(height_scz.del, aes(x=fcontrol, y=fcase, col=bmi_col)) + geom_point(alpha=0.05) + theme(legend.title=element_blank()) + guides(colour = guide_legend(override.aes = list(alpha = 1))) +facet_wrap(~cut_cnvfreq)



ggplot(height_scz.del, aes(x=beta_height, y=z, col=h_col)) + geom_point(alpha=0.06) + theme(legend.title=element_blank()) + guides(colour = guide_legend(override.aes = list(alpha = 1)))


height_scz.del$raredel <- factor(height_scz.del$fdel<0.01, levels=c(T,F), labels=c("rare del", "common del"))
height_scz.del$raredup <- factor(height_scz.del$fdup<0.01, levels=c(TRUE, FALSE), labels=c("rare dup", "common dup"))

pander(table(height_scz.del$raredel,height_scz.del$raredup))

ggplot(height_scz.del, aes(x=beta_height, y=z, col=h_col)) + geom_point(alpha=0.06) + theme(legend.title=element_blank()) + guides(colour = guide_legend(override.aes = list(alpha = 1))) + facet_grid(raredel~raredup)

ggplot(height_scz.del, aes(x=beta_bmi, y=z, col=bmi_col)) + geom_point(alpha=0.06) + theme(legend.title=element_blank()) + guides(colour = guide_legend(override.aes = list(alpha = 1))) + facet_grid(raredel~raredup)



```


Join the deletions and duplications together by row.


```{r join}
height_scz.del$scz_cnv = "del"
height_scz.dup$scz_cnv = "dup"

height_scz <- rbind(height_scz.del, height_scz.dup)

ggplot(height_scz, aes(x=fcase, y=fcontrol, col=bmi_col)) + geom_point(alpha=0.05) + theme(legend.title=element_blank()) + guides(colour = guide_legend(override.aes = list(alpha = 1))) +facet_grid(scz_cnv~cut_cnvfreq)

ggplot(height_scz, aes(x=fcase, y=fcontrol, col=h_col)) + geom_point(alpha=0.05) + theme(legend.title=element_blank()) + guides(colour = guide_legend(override.aes = list(alpha = 1))) +facet_grid(scz_cnv~cut_cnvfreq)

```
Have a look at the association of CNV control frequencies (from the height data) with the effect sizes for 
scz, BMI and height.  

```{r freq_control_height, echo=FALSE}
p1 <- ggplot(height_scz, aes(x=fdel+fdup, y=z, col=h_col)) + geom_point(alpha=0.1)+ scale_x_log10() + guides(colour = guide_legend(override.aes = list(alpha = 1)))
p2 <- ggplot(height_scz, aes(x=fdel+fdup, y=z, col=bmi_col)) + geom_point(alpha=0.1)+ scale_x_log10()+ guides(colour = guide_legend(override.aes = list(alpha = 1)))
p3 <- ggplot(height_scz, aes(x=fdel+fdup, y=beta_height, col=h_col)) + geom_point(alpha=0.05) + scale_x_log10()+ guides(colour = guide_legend(override.aes = list(alpha = 1)))
p4 <- ggplot(height_scz, aes(x=fdel+fdup, y=beta_bmi, col=bmi_col)) + geom_point(alpha=0.05) + scale_x_log10()+ guides(colour = guide_legend(override.aes = list(alpha = 1)))
grid.arrange(p1, p2, p3, p4, ncol=2, top="Height data CNV Frequencies and Effect Sizes")
```

Have a look at the association of CNV control frequencies (from the scz data) with the effect sizes for 
scz, BMI and height.  

```{r freq_control, echo=FALSE}
p1 <- ggplot(height_scz, aes(x=fcontrol+0.5, y=z, col=h_col)) + geom_point(alpha=0.1)+ scale_x_log10() + guides(colour = guide_legend(override.aes = list(alpha = 1)))
p2 <- ggplot(height_scz, aes(x=fcontrol+0.5, y=z, col=bmi_col)) + geom_point(alpha=0.1)+ scale_x_log10() + guides(colour = guide_legend(override.aes = list(alpha = 1)))
p3 <- ggplot(height_scz, aes(x=fcontrol+0.5, y=beta_height, col=h_col)) + geom_point(alpha=0.05) + scale_x_log10() + guides(colour = guide_legend(override.aes = list(alpha = 1)))
p4 <- ggplot(height_scz, aes(x=fcontrol+0.5, y=beta_bmi, col=bmi_col)) + geom_point(alpha=0.05) + scale_x_log10() + guides(colour = guide_legend(override.aes = list(alpha = 1)))
grid.arrange(p1, p2, p3, p4, ncol=2, top="SCZ data Control CNV Frequencies and Effect Sizes")
```

In this notebook I have joined the data together, and it seems that the CNVs match almost perfectly.  The height CNVs 
are not filtered to just very rare CNVs so there are more of the higher frequencies.  There are some CNVs that appear are
significant for both, and the directions that these assocate with schizophrenia 
and height and BMI are different.  It may be that 
we can do the same thing with p-values from genome wide significance tests, 
just picking the closest and it would be interesting 
but would take a while.

Why these CNV results are mapped to hg18 while the GWAS SNP results 
are always to hg19 I am not entirely sure but I can lift over the 
positions of these CNVs to hg19.



## Regions

### Duplication Region

Fill in the gaps with the full data set for these regions.  

```{r duplication_region, echo=FALSE}
if (FALSE) {
  install.load("gatepoint")
  x11()
  plot(height_scz.dup$fcontrol, height_scz.dup$fcase, col=height_scz.dup$h_col, pch=18)
  l <- fhs(data.frame(x=height_scz.dup$fcontrol, y=height_scz.dup$fcase))
  x11()
    plot(height_scz.dup$beta_height, height_scz.dup$z, col=height_scz.dup$h_col, pch=18)
  l <- fhs(data.frame(x=height_scz.dup$beta_height, y=height_scz.dup$z))
    cat("c(",paste(as.numeric(l),collapse=","),")", sep="")
  x11()
    plot(height_scz.dup$beta_bmi, height_scz.dup$z, col=height_scz.dup$h_col, pch=18)
  l <- fhs(data.frame(x=height_scz.dup$beta_bmi, y=height_scz.dup$z))
    cat("c(",paste(as.numeric(l),collapse=","),")", sep="")

}
id_dup <- c(12478,12479,12480,12481,12482,12483,12484,12485,12486,12487,12488,12489,12490,12491,12492,12493,12494,12495,12496,12497,12498,12499,12500,12501,12502,12503,12504,12505,12506,12507,12508,12509,12510,12511,12512,12513,12514,12515,12516,12517,12518,12519,12520,12521,12522,12523,12524,12525,12526,12527,12528,12529,12530,12531)
id_dup2 <- c(344,345,346,347,348,349,350,351,352,353,354,355,356,357,358,359,360,361,363,364,365,366,367,368,369,370,371,372)
id_dup3 <- c(12532,12533)
id_dup4 <- c(9515,9516,9517,9518,9520,9521)




plot_subset_regions <- function(id, target, ex=2000000) {
  opar <- par(mfrow=c(2,1), mar=c(0,3,3,2))
  test_region <- GRanges(seqnames=target$seqnames[id[1]], 
                       IRanges(start=target$start_h[id[1]], end=target$end[id[length(id)]])) 

  plot_region <- test_region + ex

  height_region <- subsetByOverlaps(gheight_cnv, plot_region)
  scz_del_region <- subsetByOverlaps(gscz.del, plot_region)
  scz_dup_region <- subsetByOverlaps(gscz.dup, plot_region)

  yrange <- range(height_region$beta_height, height_region$beta_bmi, scz_del_region$z, scz_dup_region$z)
  
  plot(start(height_region), height_region$beta_height, ylim=yrange, axes=FALSE)
    points(start(height_region), height_region$beta_bmi, col="blue")

  points(start(scz_del_region), scz_del_region$z, col="red")
  points(start(scz_dup_region), scz_dup_region$z, col="green")
  axis(2)
  opar <- par(   mar = c(3,3,0,2))

  yrange <- range(height_region$fdel/100, height_region$fdup/100, 
                  scz_dup_region$fcontrol, scz_del_region$fcontrol,
                  scz_del_region$fcase, scz_dup_region$fcase)
 
    plot(start(height_region), height_region$fdel/100, ylim=yrange, col="red", type="l", axes=FALSE, lty=3)
    lines(start(height_region), height_region$fdup/100, col="blue",lty=3)

    lines(start(scz_del_region), scz_del_region$fcontrol, col="red", lty=2)
    lines(start(scz_del_region), scz_del_region$fcase, col="red", lty=1)
    
    lines(start(scz_dup_region), scz_dup_region$fcontrol, col="blue", lty=2)
    lines(start(scz_dup_region), scz_dup_region$fcase, col="blue", lty=1)
rug(target$start_s[id], ticksize=-0.05, col="blue")
axis(1);axis(2)
}

plot_subset_regions(id_dup,height_scz.dup, ex=2000000 )
plot_subset_regions(id_dup2,height_scz.dup, ex=2000000 )
plot_subset_regions(id_dup3,height_scz.dup, ex=2000000 )
plot_subset_regions(id_dup4,height_scz.dup, ex=2000000 )



````


### Deletion Regions

```{r deletion_regions, echo=FALSE}
if (FALSE) {
  install.load("gatepoint")
  x11()
  dat <- data.frame(x=height_scz.del$beta_height, 
                    y=height_scz.del$z, 
                    col=height_scz.del$seqnames)[height_scz.del$h_col=="H&S sig",]
  plot(dat$x,dat$y,  pch=18, col=factor(dat$col))
  l <- fhs(dat)
  cat("c(",paste(as.numeric(l),collapse=","),")", sep="")
    x11()
  dat <- data.frame(x=height_scz.del$beta_bmi, 
                    y=height_scz.del$z, 
                    col=height_scz.del$seqnames)[height_scz.del$h_col=="H&S sig",]
  plot(dat$x,dat$y,  pch=18, col=factor(dat$col))
  l <- fhs(dat)
  cat("c(",paste(as.numeric(l),collapse=","),")", sep="")
}

l_del1 <- c(179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210)
l_del2 <- c(2322,2323,2324,2325,2332,2333,2334,2335,2336,2337,2338,2339)
l_del3 <- c(9567,9568,9569,9570,9571,9572,9573,9574,9575,9576,9577,9578,9579,9580,9581,9582,9583,9584,9585,9586,9587,9588,9589,9590,9591,9592,9593,9594,9595,9596,9597,9598,9599,9600,9601,9602,9603,9604,9605,9606)


plot_subset_regions(l_del1,height_scz.del, ex=2000000 )
plot_subset_regions(l_del2,height_scz.del, ex=2000000 )
plot_subset_regions(l_del3,height_scz.del, ex=2000000 )


```


## Gene Centric Analysis

I would normally do this first but want to look at how the results from the position analysis match up with
the gene centric analysis.

```{r prep_genes, echo=FALSE}
load(here("output", "genesGR.rda"))
genome(genesGR18) <- "hg18"
```

### height and SCZ

Count overlaps between CNVs and genes, then count overlaps between significant CNVs and genes. 

```{r count_overlaps_raw}
count_scz.del <- countOverlaps(genesGR18, gscz.del[gscz.del$n>5])

count_height <- countOverlaps(genesGR18, gheight_cnv)

table(height=count_height>0, scz_del =count_scz.del>0 )
```

So the total number of genes that have overlaps from height CNV are `r 2210+2103`, and hte total number from
scz_del are `r 2680+2103`.

So filter by those genes that are available in both

```{r count_overlaps_significant}
filtered_genes_dup <- genesGR18[count_height>1 & count_scz.del>1]

count_sig_scz.del <- countOverlaps(filtered_genes_dup, gscz.del[gscz.del$p<0.01])
count_height.sig <- countOverlaps(filtered_genes_dup, gheight_cnv[gheight_cnv$p_height<0.01])
tb <- table(height_sig = count_height.sig>0, scz_del_sig = count_sig_scz.del>0)
tb

chisq.test(tb)
```

Count overlaps between CNVs and genes, then count overlaps between significant CNVs and genes. 

```{r overlaps_deletion}
count_scz.del <- countOverlaps(genesGR18, gscz.dup[gscz.dup$n>5])

table(height=count_height>0, scz_del =count_scz.del>0 )
```


Filter by those genes that are available in both

```{r significant_overlaps_deletion}
filtered_genes <- genesGR18[count_height>1 & count_scz.del>1]

count_sig_scz.del <- countOverlaps(filtered_genes, gscz.del[gscz.del$p<0.01])
count_height.sig <- countOverlaps(filtered_genes, gheight_cnv[gheight_cnv$p_height<0.01])
tb <- table(height_sig = count_height.sig>0, scz_del_sig = count_sig_scz.del>0)
tb

chisq.test(tb)
```


So there apppears to be a tendency for CNVs that are associated with one to be the same as CNVs that are 
associated with the other, but it may be that I need a more sophisticated way of doing the 
significance testing, since it may just be that these are the genes where it is possible to 
get significant results for both.


We could do the same thing with SNPs. 


